### PR TITLE
Recommend use of initializers for config structs

### PR DESCRIPTION
**Pull request by Aneurysm9**
Updates style guide and changes existing configure() implementations to newConfig() as appropriate.

### CODE REVIEW

File: exporters/otlp/otlp.go [hunk 1]
```diff
@@ -59,10 +59,15 @@ type Exporter struct {
 var _ tracesdk.SpanExporter = (*Exporter)(nil)
 var _ metricsdk.Exporter = (*Exporter)(nil)
 
-func configureOptions(cfg *Config, opts ...ExporterOption) {
+func NewConfig(opts ...ExporterOption) Config {
```
Comment (MrAlias) [inline exporters/otlp/otlp.go:62]: Should this remain unexported?

Comment (Aneurysm9) [reply exporters/otlp/otlp.go:62]: The `Config` type is exported, so I figured the initializer should be as well.  Though it does seem that I need to add a docblock there.

Comment (MrAlias) [reply exporters/otlp/otlp.go:62]: > The `Config` type is exported, so I figured the initializer should be as well. Though it does seem that I need to add a docblock there.

Good point, sounds reasonable to me. +1 on the docs as well :+1:

**Review by MrAlias — APPROVED** [review]

File: exporters/otlp/otlp.go (hunk 1 above)
Comment (XSAM) [reply exporters/otlp/otlp.go:64]: All fields are unexported in `Config`, maybe we should also make the `Config` unexported? so we can remain this function unexported.

https://github.com/open-telemetry/opentelemetry-go/blob/c9726efd6075c1dccbafc5a17724f0299fd1998a/exporters/otlp/options.go#L66-L76

Comment (Aneurysm9) [reply exporters/otlp/otlp.go:64]: Excellent point.  Without any exported fields I suppose there's no value in having an exported configuration struct.  The whole point would be for users to write custom options which would need to be able to manipulate fields in the struct.

**Review by XSAM — APPROVED** [review]

**Comment by XSAM** [comment]
LGTM 🚀

**Pull request merged**